app/models/projects/acts_as_customizable_patches.rb

```diff
@@ -0,0 +1,248 @@
+      # query the available custom fields on a global level when updating custom field values
+      # in order to support implicit activation of custom fields when values are provided during an update
+      self._query_available_custom_fields_on_global_level = true
+      result = yield
+      self._query_available_custom_fields_on_global_level = false
+
+      result
+    end
+
+    def active_custom_field_ids_of_project
+      # show all project custom fields in the project creation form
+      # later on, only those with values will be activated via before_save hook `build_missing_project_custom_field_project_mappings`
+      # a persisted project will then only show the activated custom fields
+      # this approach also supports project duplication based on project templates
+      if new_record?
+        ProjectCustomField.pluck(:id)
+      else
+        project_custom_field_project_mappings.pluck(:custom_field_id)
+          .concat(ProjectCustomField.required.pluck(:id))
+          .uniq
+        # if for whatever reason a required custom field is not activated for this instance,
+        # we need to make sure it's treated as activated especially in context of the validation
+      end
+    end
+
+    def available_custom_fields(global: false)
+      # overrides acts_as_customizable
+      # in contrast to acts_as_customizable, custom_fields are enabled per project
+      # thus we need to check the project_custom_field_project_mappings
+      custom_fields = ProjectCustomField
+        .includes(:project_custom_field_section)
+
+      # available_custom_fields is called from within the acts_as_customizable module
+      # we don't want to adjust these calls, but need a way to query the available custom fields on a global level in some cases
+      # thus we pass in this parameter as an instance flag implicitly here,
+      # which is not nice but helps us to touch acts_as_customizable as little as possible
+      #
+      # additionally we provide the `global` parameter to allow querying the available custom fields on a global level
+      # when we have explicit control over the call of `available_custom_fields`
+      unless global || _query_available_custom_fields_on_global_level
+        custom_fields = custom_fields.where(id: active_custom_field_ids_of_project)
+      end
+
+      custom_fields
+    end
+
+    def available_project_custom_fields_grouped_by_section
+      sorted_available_custom_fields
+        .group_by(&:custom_field_section_id)
+    end
+
+    def available_custom_fields_by_section(section)
+      available_custom_fields
+        .where(custom_field_section_id: section.id)
+    end
+
+    def sorted_available_custom_fields
+      available_custom_fields
+        .sort_by { |pcf| [pcf.project_custom_field_section.position, pcf.position_in_custom_field_section] }
+    end
+
+    def sorted_available_custom_fields_by_section(section)
+      available_custom_fields_by_section(section)
```

Comment:
This is the only place we use the `available_custom_fields_by_section`. Both methods are just 1 liners, I would rather merge them together.
```suggestion
      available_custom_fields
        .where(custom_field_section_id: section.id)
```